python-avd/pyavd/_eos_designs/structured_config/underlay/utils.py

```diff
@@ -253,7 +253,8 @@ def _get_l2_as_subint(self: AvdStructuredConfigUnderlay, link: dict, svi: dict,
             "peer_type": link["peer_type"],
             "description": default(svi.get("description"), svi["name"]),
             "shutdown": not (svi.get("enabled", False)),
-            "type": "routed" if is_native else "l3dot1q",
+            "type": "l3dot1q" if not is_native else None,
```

Comment:
are we not removing the type key from subinterfaces?

Reply:
not handling `type: l3dot1q/l2dot1q` in this PR, it will be covered in the PR for encapsulation model